### ISSUE TITLE
Add high-level plotting function for log-odds ratios

### ISSUE THREAD

**Issue by Marius1311** [theislab/cellrank#636]
- [ ] Additional function parameters, change functionality or change defaults?
- [ ] New estimator in `cr.tl.estimators`?
- [ ] New kernel in `cr.tl.kernels`?
- [ ] New gene trend model in `cr.ul.models`?
- [ ] New plotting function in `cr.pl`?
- [ ] Other?

Esp. for time-series data, it's important to see how the probability of one fate vs. another evolves over time. For this, it would be important to have a high level plotting function that shows log-odd ratios over categorical covariates, which could be time-points but also cluster labels. For an example plot in the case of time-point annotations, see Fig. 1 below. The idea is to show the distribution of log-odds at the single cell level with the option to color in the expression of a particular gene. Comments
- have two keys, `fate_1` and `fate_2`, where each of them can be either a key to a terminal macrostate (or a combination, as in `g.compute_absorption_probabilities`) or `rest`, in which all remaining fates are considered. 
- I wouldn't allow for passing a grid of fates, that's unnecessarily complicated. I would however allow to pass a list of genes and issue one plot per gene. 

For code, see below.

## Fig. 1
![image](https://user-images.githubusercontent.com/24494650/123618396-73daf700-d808-11eb-9bcc-dca557996b43.png)

## Code
```python
def plot_fate(adata: AnnData, 
              fate: str, 
              gene: Optional[str] = None, 
              threshold: float = 0, 
              title: str = "", 
              figsize: Tuple[int] = (10, 4),
              dpi=80):
    """Utility function to plot distribution of log-odds of `fate` vs. all
    
    Log-odds are plotted as a function of the experimental time-point. Optionally, gene 
    expression can be overlaid to check for genes that show early fate bias. 
    
    Parameters
    --------
    adata
        Annodated data matrix.
    fate
        Key from `adata.obsm['to_terminal_states'].names`.
    gene
        Key from `adata.var_names`.
    title
        Title of the plot.
    figsize
        Figure size
    dpi
        Figure resolution
        
    Returns
    -------
    Nothing, just plots. 
    """

    # define log-odds
    fate1 = adata.obsm['to_terminal_states'][fate].X
    fate2 = 1 - fate1
    log_odds = np.log(1e-9 + np.divide(fate1, fate2, out=np.zeros_like(fate1), where=fate2 != 0))
    adata.obs['log_odds'] = log_odds
    
    # define keys to be pulled from adata
    keys = ['day', 'log_odds']
    
    # optionally, add gene expression
    if gene != None:
            
        # threshold gene expression
        adata.obs[f"{gene}+"] = (adata[:, gene].X.A > threshold).flatten()
        keys.append(f"{gene}+") 

    # get correponding dataframe
    plotting_df = sc.get.obs_df(adata, keys=keys)

    # filter out cells where either fate is zero
    zero_mask = np.logical_or(fate1 == 0, fate2 == 0)
    plotting_df = plotting_df[~zero_mask]
    
    # prepare for plotting
    figure = plt.figure(figsize=figsize, dpi=dpi)
    ax = figure.add_subplot(111)

    # use seaborn here for some jiiter
    if gene != None:
        ax = sns.stripplot('day', 'log_odds', data=plotting_df[~plotting_df[f"{gene}+"]], jitter=0.4, 
                           color='k', ax=ax, s=0.5)
        ax = sns.stripplot('day', 'log_odds', data=plotting_df[plotting_df[f"{gene}+"]], jitter=0.4, 
                           color='r', ax=ax, s=2)
    else:
        ax = sns.stripplot('day', 'log_odds', data=plotting_df, jitter=0.4, 
                           color='k', ax=ax, s=2)
            
    ticks = np.arange(0, len(plotting_df['day'].cat.categories), 4)
    ax.set_xticks(ticks=ticks)

    ax.set_xlabel('Day')
    ax.set_ylabel('Log Odds') 
    plt.title(title)

    plt.show()
```

**Comment by Marius1311** [comment]
I would rather try to do this as fast as possible now and skip pseudotime binning, we can add this later.

**Comment by michalk8** [comment]
> I would rather try to do this as fast as possible now and skip pseudotime binning, we can add this later.

Agreed, it's also very easy to do for the user as `pandas.cut(pt, bins=...)`.

**Comment by michalk8** [comment]
closed via #642
